splunk_add_on_ucc_framework/__init__.py

```diff
@@ -310,23 +313,29 @@ def _install_libs(requirements, ucc_target, installer="python3"):
     else:
         logging.info(f"  Not using common requirements")    
 
-    if os.path.exists(os.path.join(path,"lib","py2", "requirements.txt")):
-        logging.info(f"  Uses py2 requirements")    
-        _install_libs(requirements=os.path.join(path,"lib","py2", "requirements.txt"), installer="python2", ucc_target=os.path.join(ucc_lib_target, "py2"))
-    elif os.path.exists(os.path.join(os.path.abspath(os.path.join(path, os.pardir)), "requirements_py2.txt")):
-        logging.info(f"  Uses py2 requirements")    
-        _install_libs(requirements=os.path.join(os.path.abspath(os.path.join(path, os.pardir)), "requirements_py2.txt"), installer="python2", ucc_target=os.path.join(ucc_lib_target, "py2"))        
-    else:
-        logging.info(f"  Not using py2 requirements")    
-
-    if os.path.exists(os.path.join(path, "lib","py3","requirements.txt")):
-        logging.info(f"  Uses py3 requirements")            
-        _install_libs(requirements=os.path.join(path,"lib", "py3","requirements.txt"), ucc_target=os.path.join(ucc_lib_target, "py3"))
-    elif os.path.exists(os.path.join(os.path.abspath(os.path.join(path, os.pardir)), "requirements_py3.txt")):
-        logging.info(f"  Uses py3 requirements")    
-        _install_libs(requirements=os.path.join(os.path.abspath(os.path.join(path, os.pardir)), "requirements_py3.txt"), installer="python3", ucc_target=os.path.join(ucc_lib_target, "py2"))        
-    else:
-        logging.info(f"  Not using py3 requirements")    
+
+    #Prevent certain packages from being included pip could be dangerous others are just wasted space
+    noshipdirs = ['setuptools', 'bin', 'pip', 'distribute', 'wheel']
+    p = Path(ucc_lib_target)
+    try:
+        for nsd in noshipdirs:
+            for o in p.glob(nsd + '*'):
+                if o.is_dir():
+                    shutil.rmtree(o)
+    except FileNotFoundError:
+        pass
+
+    #Remove execute bit from any object in lib
+    NO_USER_EXEC = ~stat.S_IEXEC
+    NO_GROUP_EXEC = ~stat.S_IXGRP
+    NO_OTHER_EXEC = ~stat.S_IXOTH
+    NO_EXEC = NO_USER_EXEC & NO_GROUP_EXEC & NO_OTHER_EXEC
+
+    for o in p.rglob("*"):
+        if not o.is_dir() and os.access(o, os.X_OK):
+            logging.info(f"  fixing {o} execute bit")   
+            current_permissions = stat.S_IMODE(os.lstat(o).st_mode)
+            os.chmod(o, current_permissions & NO_EXEC)
```

Comment:
I tested it `meraki` library, it says
```
  fixing /Users/arys/src/ucc-test-dir/output/Splunk_TA_UCCExample/lib/aiohttp/_http_writer.cpython-37m-darwin.so execute bit
  fixing /Users/arys/src/ucc-test-dir/output/Splunk_TA_UCCExample/lib/aiohttp/_helpers.cpython-37m-darwin.so execute bit
  fixing /Users/arys/src/ucc-test-dir/output/Splunk_TA_UCCExample/lib/aiohttp/_frozenlist.cpython-37m-darwin.so execute bit
  fixing /Users/arys/src/ucc-test-dir/output/Splunk_TA_UCCExample/lib/aiohttp/_websocket.cpython-37m-darwin.so execute bit
  fixing /Users/arys/src/ucc-test-dir/output/Splunk_TA_UCCExample/lib/aiohttp/_http_parser.cpython-37m-darwin.so execute bit
  fixing /Users/arys/src/ucc-test-dir/output/Splunk_TA_UCCExample/lib/multidict/_multidict.cpython-37m-darwin.so execute bit
  fixing /Users/arys/src/ucc-test-dir/output/Splunk_TA_UCCExample/lib/yarl/_quoting_c.cpython-37m-darwin.so execute bit
  ```
 and I am not sure that it is not going to break anything in the add-on.
  
We can include executable check in CI for Splunk libraries, like "splunktalib", "splunktaucclib", etc. And we can leave specific cases (like `meraki`) to the add-on developers.

Reply:
This is safe for Meraki no so file needs execute on Linux after kernel 2.0